packages/samples/headless-react/src/components/smart-snippet/smart-snippet.class.tsx

```diff
@@ -0,0 +1,104 @@
+import {Component, ContextType} from 'react';
+import {
+  buildSmartSnippet,
+  SmartSnippet as HeadlessSmartSnippet,
+  SmartSnippetState,
+  Unsubscribe,
+} from '@coveo/headless';
+import {AppContext} from '../../context/engine';
+
+export class SmartSnippet extends Component<{}, SmartSnippetState> {
+  static contextType = AppContext;
+  context!: ContextType<typeof AppContext>;
+
+  private controller!: HeadlessSmartSnippet;
+  private unsubscribe: Unsubscribe = () => {};
+
+  componentDidMount() {
+    this.controller = buildSmartSnippet(this.context.engine!);
+    this.updateState();
+
+    this.unsubscribe = this.controller.subscribe(() => this.updateState());
+  }
+
+  componentWillUnmount() {
+    this.unsubscribe();
+  }
+
+  private updateState() {
+    this.setState(this.controller.state);
+  }
+
+  private answerStyles(expanded: boolean) {
+    const maskImage = () =>
+      expanded
+        ? 'none'
+        : 'linear-gradient(to bottom, black 50%, transparent 100%)';
+
+    return {
+      maxHeight: expanded ? '100%' : '100px',
+      maxWidth: '200px',
+      overflow: 'hidden',
+      marginBottom: '10px',
+      maskImage: maskImage(),
+      WebkitMaskImage: maskImage(),
+    };
```

Comment:
What is the mask image for?

Reply:
Adds a "blur" effect at the bottom of the answer when it is collapsed, to give a better sense that "this can be expanded".